landlab/components/soil_grading/soil_grading.py

```diff
@@ -0,0 +1,540 @@
+            lower_limit = maxsize * (1 / self._N) ** (power_of)
+            return lower_limit
+
+        upperlimits = []
+        lowerlimits = []
+        num_of_size_classes_plusone = self._n_sizes + 1
+
+        if input_sizes_flag:
+            meansizes = self._meansizes
+            self._meansizes = np.array(meansizes)
+            self._upperlims = np.array(meansizes)
+            self._lowerlims = np.insert(np.array(meansizes), 0, 0)
+        else:
+            maxsize = self._grain_max_size
+            power_of = self._power_of
+            for _ in range(num_of_size_classes_plusone - 1):
+                upperlimits.append(maxsize)
+                maxsize = lower_limit_of(maxsize)
+                lowerlimits.append(maxsize)
+
+            self._upperlims = np.sort(upperlimits)
+            self._lowerlims = np.sort(lowerlimits)
+            self._meansizes = (
+                np.array(self._upperlims) + np.array(self._lowerlims)
+            ) / 2
+
+        self._grid.at_node["grains_classes__size"][self._grid.nodes] *= self._meansizes
+
+    def create_dist(
+        self, median_size=None, std=None, is_bedrock_distribution_flag=False
+    ):
+        """
+        This procedure split the total soil weight between all size classes
+        assuming normal distribution around the median size class. Different
+        grain size distribution can be assigned to the initial soil layer and
+        the bedrock layer
+        """
+
+        if not is_bedrock_distribution_flag:
+
+            is_bedrock_distribution_flag = self._is_bedrock_distribution_flag
+            median_size = self._initial_median_size
+            std = self._std
+            total_soil_weight = self._initial_total_soil_weight
+            grains_weight__distribution = self._generate_normal_distribution(
+                median_size=median_size, std=std, total_soil_weight=total_soil_weight
+            )
+
+        else:
+            total_bedrock_weight = 10000  # There is no meaning for this number
+            if median_size is None:
+                grains_weight__distribution = self.g_state0
+            else:
+                median_size = median_size
+                if std is None:
+                    std = self._CV * median_size
+                else:
+                    std = std
+                grains_weight__distribution = self._generate_normal_distribution(
+                    median_size=median_size,
+                    std=std,
+                    total_soil_weight=total_bedrock_weight,
+                )
+
+        if not is_bedrock_distribution_flag:
+            self.g_state = (
+                np.ones(
+                    (
+                        int(np.shape(self._grid)[0]),
+                        int(np.shape(self._grid)[1]),
+                        int(len(grains_weight__distribution)),
+                    )
+                )
+                * grains_weight__distribution
+            )
```

Comment:
I think you can rewrite this more succinctly as,
```python
self.g_state = np.full(
    self.grid.shape + (len(grains_weight__distribution),), grains_weight__distribution
)
```

Reply:
Replaced.